.github/workflows/pre-commit.yml

```diff
@@ -0,0 +1,13 @@
+name: pre-commit
+
+on: [push, pull_request]
+
+jobs:
+  pre-commit:
+    runs-on: ubuntu-latest
+    steps:
```

Comment:
I think you forgot to install build dependencies.

Reply:
pre-commit pulls in all of the dependencies you need to run its checks (not only in github actions but also when you use it locally), and then caches them using `actions/cache@v3`.

Reply:
Oh do you mean?:

```
- name: Get required packages
  run: sudo apt-get install libsensors-dev libssl-dev libpulse-dev libnotmuch-dev
```

Reply:
I mean dependencies of `i3status-rust`, `sudo apt-get install libsensors-dev libssl-dev libpulse-dev libnotmuch-dev`.